src/types/input.ts

```diff
@@ -1,11 +1,19 @@
-export interface IInputField {
-  id: string,
-  required?: boolean,
-  placeholder: string,
-  labelText: string,
-  error: boolean,
+import { FieldInputProps, FormikProps } from 'formik';
+
+interface ICreateProductFormValues {
+  product_name: string,
+  product_image: string,
+  product_details: string,
+  product_price: string,
+}
+
+export interface IOtherProps {
+  placeholder?: string,
+  autocomplete?: string,
+  label?: string,
 }
 
-export interface IStyledInputField {
-  required?: boolean,
-};
+export interface IInputField {
+  field: FieldInputProps<string>,
+  form: FormikProps<ICreateProductFormValues>,
```

Comment:
Looks like general `IInputField` depends on concrete  `ICreateProductFormValues`
for now just add `FormikProps<any>`